### PR TITLE
(FM-3924) Create Master/Agent automation test for Puppetlabs-motd

### PR DESCRIPTION
**Pull request by phongdly**
This PR has one test case of creating /etc/motd file using content parameter, the test case is located at puppetlabs-motd/tests/beaker/tests/01_create_motd_content.rb

The PR also uses Test Organization in Module Repositories as specified in https://confluence.puppetlabs.com/pages/viewpage.action?spaceKey=ENG&title=Test+Tiering+for+the+Modules+Team document written by @cowofevil

### CODE REVIEW

**Comment by cowofevil** [comment]
Why is there so many Beaker host files included? I feel like we could use a much smaller subset.

File: Gemfile [hunk 1]
```diff
@@ -22,4 +22,18 @@ else
...
+    gem 'beaker-rspec', *location_for(beaker_rspec_version)
+  else
+    gem 'beaker-rspec',  :require => false
+  end
+  gem 'serverspec',    :require => false
+  gem 'beaker-puppet_install_helper', :require => false
+  gem 'master_manipulator', '1.1.2', :require => false
+end
+
 # vim:ft=ruby
```
Comment (cowofevil) [inline Gemfile]: I think this module is uses module-sync so you should make changes directly to the Gem file.

File: tests/beaker/pre-suite/00_pe_install.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require 'master_manipulator'
+test_name 'FM-3804 - C61700 - Install Puppet Enterprise'
```
Comment (cowofevil) [inline tests/beaker/pre-suite/00_pe_install.rb]: The ```FM-3804``` ticket referenced here is associated with Websphere.

File: tests/beaker/pre-suite/01_motd_module_install.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+test_name 'FM-3918 - C95871 - Install MOTD module'
+
+step 'Install MOTD Module'
+proj_root = File.expand_path(File.join(File.dirname(__FILE__), '../../../'))
+staging = { :module_name => 'puppetlabs-motd' }
+local = { :module_name => 'motd', :source => proj_root, :target_module_path => master['distmoduledir'] }
+
+# in CI install from staging forge, otherwise from local
+install_dev_puppet_module_on(master, options[:forge_host] ? staging : local)
```
Comment (cowofevil) [inline tests/beaker/pre-suite/01_motd_module_install.rb]: Missing newline at EOF.

File: tests/beaker/test_run_scripts/upgrade_from_2015_2_3_to_Ankeny.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+bundle install --without build development test --path .bundle/gems
+
+bundle exec beaker \
+  --no-provision \
+  --host log/latest/hosts_preserved.yml \
+  --debug \
+  --pre-suite setup/upgrade.rb \
+  --test acceptance/tests/file-sync/file-sync_commit.rb \
+  --keyfile ~/.ssh/id_rsa-acceptance \
+  --load-path lib
```
Comment (cowofevil) [inline tests/beaker/test_run_scripts/upgrade_from_2015_2_3_to_Ankeny.sh]: This looks weird, what is the purpose of this script?

File: tests/beaker/tests/01_create_motd_content.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+inject_site_pp(master, get_site_pp_path(master), site_pp)
+
+step 'Run Puppet Agent to apply the above manifest'
+confine_block(:except, :roles => %w{master dashboard database}) do
+  agents.each do |agent|
+    on(agent, puppet('agent -t --environment production'), :acceptable_exit_codes => [0,2]) do |result|
+      assert_no_match(/Error:/, result.stderr, 'Unexpected error was detected!')
+    end
+  end
+end
```
Comment (cowofevil) [inline tests/beaker/tests/01_create_motd_content.rb]: You fail to verify that the contents of the MOTD file are correct after the Puppet run.

**Comment by cowofevil** [comment]
You should add a README for the tests folder with some information about running the tests. Especially given the fact that the tests are only runnable inside Puppet Labs.

File: tests/beaker/pre-suite/00_pe_install.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+# Init
+step 'Install PE'
+install_pe
```
Comment (cowofevil) [inline tests/beaker/pre-suite/00_pe_install.rb]: There is no requirement that PE be used for the master. You could use the FOSS stuff instead.

File: tests/beaker/test_run_scripts/upgrade_from_2015_2_3_to_Ankeny.sh (hunk 4 above)
Comment (phongdly) [inline tests/beaker/test_run_scripts/upgrade_from_2015_2_3_to_Ankeny.sh]: This is a script that I used in integration testing, copy and paste typo, will remove it in the next push

File: Gemfile [hunk 7]
```diff
@@ -26,4 +26,18 @@ if RUBY_VERSION < '2.0'
...
+group :system_tests do
+  if beaker_version = ENV['BEAKER_VERSION']
+    gem 'beaker', *location_for(beaker_version)
+  end
+  if beaker_rspec_version = ENV['BEAKER_RSPEC_VERSION']
+    gem 'beaker-rspec', *location_for(beaker_rspec_version)
+  else
+    gem 'beaker-rspec',  :require => false
+  end
+  gem 'serverspec',    :require => false
```
Comment (cowofevil) [inline Gemfile]: You have duplicate entry for "serverspec". Only specify the dependency one time.

File: Gemfile [hunk 8]
```diff
@@ -26,4 +26,18 @@ if RUBY_VERSION < '2.0'
   gem 'mime-types', '<3.0', :require => false
 end
 
+group :system_tests do
+  if beaker_version = ENV['BEAKER_VERSION']
+    gem 'beaker', *location_for(beaker_version)
+  end
+  if beaker_rspec_version = ENV['BEAKER_RSPEC_VERSION']
+    gem 'beaker-rspec', *location_for(beaker_rspec_version)
```
Comment (cowofevil) [inline Gemfile]: You have duplicate entry for "beaker-rspec". Only specify the dependency one time.

**Pull request merged**